uint64 timestamp
uint64 send_time_ns
uint8[] data
